crates/parser/src/config/mod.rs

```diff
@@ -306,41 +307,69 @@ pub enum Format {
     /// https://www.w3.org/TR/WD-logfile.html
     #[serde(rename = "w3cExtendedLog")]
     #[schemars(title = "W3C Extended Log")]
-    W3cExtendedLog(EmptyConfig),
+    W3cExtendedLog,
 }
 
 impl fmt::Display for Format {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let s = match self {
-            Format::Auto(_) => "auto",
-            Format::Avro(_) => "avro",
-            Format::Json(_) => "json",
+            Format::Auto => "auto",
+            Format::Avro => "avro",
+            Format::Json => "json",
             Format::Csv(_) => "csv",
-            Format::W3cExtendedLog(_) => "w3cExtendedLog",
+            Format::W3cExtendedLog => "w3cExtendedLog",
         };
         f.write_str(s)
     }
 }
 
 impl Default for Format {
     fn default() -> Format {
-        Format::Auto(EmptyConfig)
+        Format::Auto
     }
 }
 
 impl Format {
     pub fn is_auto(&self) -> bool {
         match self {
-            Format::Auto(_) => true,
+            Format::Auto => true,
             _ => false,
         }
     }
     pub fn non_auto(&self) -> Option<Format> {
         match self {
-            Format::Auto(_) => None,
+            Format::Auto => None,
             _ => Some(self.clone()),
         }
     }
+
+    /// Customizes the generated schema for Format to make it work better in the UI.
+    /// Format still derives `JsonSchema`, and this function delegates to that impl.
+    /// This function is used by setting `schema_with` on the property in ParseConfig.
+    fn schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        let mut schema = gen.subschema_for::<Format>().into_object();
+        // The UI doesn't handle the case where a schema specifies both `type` and `const`,
+        // and technically it's redundant to have both. So we remove the `type` property from the
+        // `type` of each variant.
+        for subschema in schema.subschemas().one_of.as_mut().unwrap().iter_mut() {
```

Comment:
Is this `unwrap()` safe?

Reply:
I think we can use:
```
unwrap_or(vec![])
```

Reply:
Yeah, this is a safe unwrap since `Format` will always have a `oneOf` schema. We want this to panic if that changes for any reason (it would do so when running unit tests).